MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/activity/login/LoginActivity.kt

```diff
@@ -0,0 +1,167 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (intent.hasExtra("user") && intent.hasExtra("password")) {
+            Log.d(TAG, "onCreate: Found benchmark category")
+            intent.extras?.run {
+                Log.d(TAG, "onCreate: Using benchmark userdata")
+                val userName = getString("user", "")
+                val password = getString("password", "")
+                sampleViewModel.login(userName, password)
+            }
+        } else {
+            sampleViewModel.loadAppLogin()
+        }
+        setContent {
+            LoginScreen()
+        }
+    }
+
+    @Composable
+    fun LoginScreen() {
+        fullyDrawnReporter.addReporter()
+
+        var userName by remember {
+            mutableStateOf(sampleViewModel.login.userName)
+        }
+        var password by remember {
+            mutableStateOf(sampleViewModel.login.password)
+        }
+        MaterialTheme {
+            Box(
+                modifier = Modifier.semantics {
+                    // Allows to use testTag() for UiAutomator's resource-id.
+                    // It can be enabled high in the compose hierarchy,
+                    // so that it's enabled for the whole subtree
+                    testTagsAsResourceId = true
+                }
+            ) {
+
+                Column(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    TopAppBar(title = {
+                        Text(text = "Please login to this sample")
+                    })
+                    InputField(
+                        title = "User: ",
+                        value = userName,
+                        tag = "userName",
+                        onValueChange = { userName = it })
+                    InputField(
+                        title = "Password: ",
+                        value = password,
+                        tag = "password",
+                        onValueChange = {
+                            password = it
+                        },
+                        visualTransformation = PasswordVisualTransformation(),
+                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password)
+                    )
+                    Button(
+                        modifier = Modifier.testTag("login"),
+                        enabled = userName.isNotEmpty() && password.isNotEmpty(),
+                        onClick = {
+                            sampleViewModel.login(userName, password)
+                            startActivity(
+                                Intent(
+                                    this@LoginActivity,
+                                    MainActivity::class.java
+                                )
+                            )
+                            finish()
```

Comment:
You have async code in the `login` method, so this should wait until login is finished. 
I guess there are 2 options:
1. we could have suspend login and start an effect here (ugly)
2. do it properly with UI events https://developer.android.com/topic/architecture/ui-layer/events#navigation-events
3. (don't care, because it's a sample 😄 )

Reply:
1 it is for now.